### ISSUE TITLE
[BUG] Fail upgrades/ mixed cluster for ES version > 7.10.2

### ISSUE THREAD

**Issue by shwetathareja** [opensearch-project/OpenSearch#806]
**Describe the bug**
OpenSearch is BWC with ES <= 7.10.2 (until 6.8). So, there should be relevant checks that prevent transport handshake and connection with Elasticsearch nodes if the version is > 7.10.2

**Comment by dblock** [comment]
@shwetathareja What's the technical reason for not supporting < 6.8?

**Comment by shwetathareja** [comment]
OpenSearch codebase still follows Elasticsearch philosophy of upgrades from last minor version of last major version released, so technically for OpenSearch being fork of 7.10.2, the minimum allowed version for upgrade is 6.8. The upgrade from ES 6.x (other than 6.8) to OpenSearch 1.0 is untested flow and there could be backward incompatibilities and failure during upgrade. 
The rest call to localhost:9200 in OpenSearch also returns minimum supported version for backward compatibility is 6.8.
So, ES 6.x (other than 6.8) is not supported for upgrade to Elasticsearch 7.x version and hence not supported in OpenSearch as well.

**Comment by elfisher** [comment]
It sounds like the upgrade path is multi-step, but still supported. 

1. Rolling upgrade to 6.8
2. Rolling upgrade to OpenSearch

Is this the same with 5.x?

1. Rolling to 5.6
2. Rolling to 6.8
3. Rolling to OpenSearch

**Comment by shwetathareja** [comment]
@elfisher  Yes, there is no direct upgrade from 6.x (except 6.8) to OpenSearch but still supported via 2 step as you called out.

As far as 5.x is concerned - the steps you mentioned should work. The catch is minimum supported index version for OpenSearch is 6.0. So, all the 5.x indices have to re-indexed before upgrade to OpenSearch.

**Comment by dblock** [comment]
@shwetathareja If you have a moment, could you please check out and comment on https://github.com/opensearch-project/project-website/pull/103 to make sure we didn't miss anything in these version stories ... should we add 5.x there too?

**Comment by nknize** [comment]
I wouldn't characterize this as a bug since we do not support "upgrade" from versions > 7.10.2 to begin with. It's part of the re-basing effort to non ES version and giving users a polite deprecation and error messaging to assist in that process is certainly an enhancement to that logic. So let's re-label as an enhancement so it's clear this is not a blocker for the RC. 



> there should be relevant checks that prevent transport handshake and connection with Elasticsearch nodes if the version is > 7.10.2

+1 for providing these useful error messages to reinforce the fork from 7.10.2.








> minimum supported index version for OpenSearch is 6.0

Correct. Anyone running ES in production should be aware of this migration path strategy put in place by Elastic as it's well documented. We need to address this as a whole to include not just the handshake / node join process but also the API transport layer for forward compatibility.

**Comment by nknize** [comment]
Unless there's a blocking reason to include this (e.g., plugins are broke) I'd bump this to the 1.1 release. Any objection?

**Comment by dblock** [comment]
> Unless there's a blocking reason to include this (e.g., plugins are broke) I'd bump this to the 1.1 release. Any objection?

None. Done. @shwetathareja feel free to self-assign

**Comment by shwetathareja** [comment]
@nknize @dblock Sounds good. I have assigned it to myself.

**Comment by minalsha** [comment]
@shwetathareja  are you working on this currently?

**Comment by anasalkouz** [comment]
Hi @shwetathareja , are you actively working on this? could you please update the issue?

**Comment by shwetathareja** [comment]
@anasalkouz @minalsha : I havn't started on it. Feel free to assign it to someone else.
